src/parsimmon.js

```diff
@@ -232,9 +232,13 @@ function encodedString(encoding, length) {
   });
 }
 
+function isInteger(value) {
+  return typeof value === "number" && Math.floor(value) === value;
+}
+
 function uintBE(length) {
-  if (length < 0 || length > 6) {
-    throw new Error("uintBE requires length in range [0, 6].");
+  if ((isInteger(length) && length < 0) || length > 6) {
+    throw new Error("uintBE requires integer length in range [0, 6].");
```

Comment:
I think the parentheses are in the wrong spot on this line. Also it might be nice to put this entire `if` condition into a helper function since it's the same in several locations

Reply:
I'll put this into a helper function.  The parens were inserted by the style fixer I think, probably to emphasize the actual associativity, which is wrong, and I am going to fix.